packages/commonwealth/client/scripts/views/pages/chain_entity_link_redirect.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import app from 'state';
+import { PageLoading } from './loading';
+import { useCommonNavigate } from 'navigation/helpers';
+
+type ChainEntityLinkRedirectProps = {
+  identifier: string;
+  scope: string;
+};
+
+export default function ChainEntityLinkRedirect({ identifier, scope }: ChainEntityLinkRedirectProps) {
+  const navigate = useCommonNavigate();
+
+  useEffect(() => {
+    const fetchChainEntityData = async () => {
+      try {
+        // 1. make query to chain events to get the specific entity data
+        const entity = await app.chainEntities.getOneEntity(scope, identifier);
+
+        // 2. query data to construct new link
+        const { title, type, typeId } = entity;
+        const newLink = {
+          source: 'proposal',
```

Comment:
we should you enum here but it is on my PR https://github.com/hicommonwealth/commonwealth/pull/3628/files#diff-65f6a84029354b4ea2d8a68ac1b9453ee76725ba26af0e19af0e1f582b9083d2R28

Reply:
Will update post-merge.